File: VB/12_Tables/TraverseThroughCells.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Presentation
Imports System.IO

Namespace TraverseThroughCells
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create a PowerPonit document.
			Dim presentation As New Presentation()

			'Load the file from disk.
			presentation.LoadFromFile("..\..\..\..\..\..\Data\Template_Ppt_1.pptx")

			Dim content As New StringBuilder()
			content.AppendLine("The data in cells of this PowerPoint file is: ")

			'Get the table.
			Dim table As ITable = Nothing
			For Each shape As IShape In presentation.Slides(0).Shapes
				If TypeOf shape Is ITable Then
					table = CType(shape, ITable)

					'Traverse through the cells of table.
					For Each row As TableRow In table.TableRows
						For Each cell As Cell In row
							content.AppendLine(cell.TextFrame.Text)
						Next cell
						content.AppendLine(vbLf)
					Next row
				End If
			Next shape

			Dim result As String = "Result-TraverseThroughCells.txt"

			'Save to file.
			File.WriteAllText(result, content.ToString())

			'Launch the file.
			PptDocumentViewer(result)
		End Sub

		Private Sub PptDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/12_Tables/TraverseThroughCells.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Presentation;
using System.IO;

namespace TraverseThroughCells
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a PowerPonit document.
            Presentation presentation = new Presentation();

            //Load the file from disk.
            presentation.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Ppt_1.pptx");

            StringBuilder content = new StringBuilder();
            content.AppendLine("The data in cells of this PowerPoint file is: ");

            //Get the table.
            ITable table = null;
            foreach (IShape shape in presentation.Slides[0].Shapes)
            {
                if (shape is ITable)
                {
                    table = (ITable)shape;

                    //Traverse through the cells of table.
                    foreach (TableRow row in table.TableRows)
                    {
                        foreach (Cell cell in row)
                        {
                            content.AppendLine(cell.TextFrame.Text);
                        }
                        content.AppendLine("\n");
                    }
                }
            }

            String result = "Result-TraverseThroughCells.txt";

            //Save to file.
            File.WriteAllText(result, content.ToString());

            //Launch the file.
            PptDocumentViewer(result);
        }

        private void PptDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}